### PR TITLE
fix: harden self-email loop filter against aliases, plus-addressing, and send-as variations

### LINKED ISSUE #37: Harden self-email loop filter against aliases, plus-addressing, and send-as variations
## Background

The email adapter filters self-sent messages by comparing `msg.from[0]?.email` against a single `NYLAS_SELF_EMAIL` config value (case-insensitive). This has edge cases:

- **Aliases**: If the email provider rewrites the From address (e.g., `nathancuria@gmail.com` → `nathan.curia@gmail.com`)
- **Plus-addressing**: `nathan+test@domain.com` vs `nathan@domain.com`
- **Send-as**: If the Nylas account has a different "send as" address configured
- **Single value**: Only one self-email is supported; multiple aliases would need multiple values

## Risk

If a self-sent email bypasses the filter, the system processes its own reply as a new inbound message, triggering another reply, creating an infinite email loop. The case-insensitive comparison (fixed in PR #34) mitigates the most common case, but alias/plus-address variations remain.

## Recommended fixes

1. Support a comma-separated list of self-email addresses (not just one)
2. Normalize both addresses (strip plus-addressing before comparison)
3. Additionally filter by checking if the message's `folders` array includes "sent" or "drafts"
4. Track recently-sent Nylas message IDs and filter them from inbound polls

## Priority

Low urgency for single-account setup. Higher urgency if multiple send-as aliases are configured.

## Context

Found during security review of PR #34.

### PR DESCRIPTION
**Pull request by josephfung**
## Summary

- **Three-layer loop guard** in the inbound email poll — folder membership, sent-message-ID tracking, and plus-address-normalized address comparison — replacing the single `selfEmail.toLowerCase()` check
- **No config changes required**: all protection is pattern-based and in-memory
- Closes #37

## What changed

**`src/channels/email/email-adapter.ts`**

- Added `normalizeForComparison(email)` helper — strips plus-addressing (`curia+tag@example.com` → `curia@example.com`) and lowercases; applied to every `selfEmail` comparison in `poll()`, `sendOutboundReply()`, and `extractParticipants()`
- Layer 1 (folder check): skip any message whose `folders` array contains `"sent"` or `"draft"` (case-insensitive substring); catches send-as alias replies regardless of From address
- Layer 2 (sent-ID tracking): after each successful `outboundGateway.send()`, the returned Nylas message ID is stored in a bounded in-memory `Set` (max 200); inbound poll skips any message whose ID is in the set
- Warn log when `fromEmail` is absent (pre-existing silent pass-through now observable)
- Warn log when `gateway.send()` succeeds but returns no `messageId` (Layer 2 gap made visible)

**`tests/unit/channels/email/email-adapter.test.ts`**

13 new tests covering all three layers, including the `security@meetcuria.com` forwarding scenario.

## Test plan

- [ ] `npm test -- tests/unit/channels/email/email-adapter.test.ts` — 39 tests, all green
- [ ] Full suite — no regressions (3 pre-existing failures unrelated to this PR)
- [ ] `npm run typecheck` — clean

## Release Notes

* **Bug Fixes**
  * Strengthened email loop prevention during inbound polling to reduce infinite reply scenarios with send-as alias forwarding. The system now uses multiple independent detection layers including folder membership checks, message ID tracking, and normalized address comparison (supporting plus-addresses and case variations) to identify and skip self-sent emails.

[Review Change Stack](https://app.coderabbit.ai/change-stack/josephfung/curia/pull/586)

### CODE REVIEW

**Comment by josephfung** [comment]
@coderabbitai review

File: CHANGELOG.md [hunk 1]
```diff
@@ -13,6 +13,10 @@ bus event types) are noted explicitly even in the `0.x` range.
 
 ## [Unreleased]
 
+### Fixed
+
+- **Self-email loop filter hardened** — inbound poll now rejects messages via three independent checks: folder membership ("sent"/"draft"), recently-sent Nylas message ID, and plus-address-normalized `selfEmail` comparison. Prevents infinite reply loops when using send-as alias addresses (e.g. `security@meetcuria.com` forwarding to the Nylas account). (#37)
```
[the remark this answers is not in the document]
Comment (josephfung) [reply CHANGELOG.md]: Fixed — condensed to one clause (~15 words after the em-dash) in a9198a8.

File: tests/unit/channels/email/email-adapter.test.ts [hunk 2]
```diff
@@ -798,6 +798,196 @@ describe('EmailAdapter — sendWithGatedDraftFallback gated-fallback', () => {
...
+    (mocks.outboundGateway.listEmailMessages as ReturnType<typeof vi.fn>).mockResolvedValueOnce([sentMsgAsInbound]);
+
+    // Reset publish spy so we only see events from this poll
+    (mocks.bus.publish as ReturnType<typeof vi.fn>).mockClear();
+    await (adapter as unknown as { poll(): Promise<void> }).poll();
+
+    expect(wasPublished(mocks)).toBe(false);
+
+    await adapter.stop();
+  });
```
[the remark this answers is not in the document]
Comment (josephfung) [reply tests/unit/channels/email/email-adapter.test.ts:937]: Fixed — added `await flushPoll()` after `adapter.start()` in a9198a8. Also added three provider-specific folder name tests (`[Gmail]/Sent Mail`, `Sent Items`, `Drafts_Old`) in the same commit to validate the substring match works for real-world provider formats.

**Comment by josephfung** [comment]
@coderabbitai approve
